index.rst

```diff
@@ -1,61 +1,383 @@
+Unpublished astronomical research, while confidential, does not have the same appeal to attackers.
+Therefore, targeted attacks by sophisticated attackers looking for data of monetary or political value are unlikely.
+
+By design, the Rubin Science Platform is open to a broad array of scientists.
+Authentication will be done via federated identity, removing the risk of storing and managing user passwords or other authentication credentials but increasing the risk of a user account compromise due to a compromise of some external authentication system outside of the control of Rubin Observatory.
+We therefore expect incidents where the account of a legitimate user of the Rubin Science Platform will be compromised, and want to provide some protection to other users of the platform against malicious activity from a compromised account.
+
+The individual applications that make up the Rubin Science Platform will be written by a variety of people and institutions with varying levels of attention to security.
+Therefore, we do not want to treat all applications contained in the Rubin Science Platform as equally trustworthy.
+
+The implication for web security is to focus efforts on providing a robust authentication layer, trust domain isolation between applications, and protection against typical XSS and CSRF attacks that an attacker might easily discover with an automated tool.
+
+For considerably more discussion of the threat model, see `SQR-041`_.
+
+.. _SQR-041: https://sqr-041.lsst.io/
+
+Authentication
+==============
+
+The Rubin Science Platform uses opaque bearer tokens for authentication.
+Each token is associated with a list of scopes, which restrict what services that token can be used to access.
+See `SQR-044`_ for the identity management requirements, including access tokens, and `SQR-049`_ for the token system design.
+`SQR-039`_ has some additional history and discussion.
+
+.. _SQR-044: https://sqr-044.lsst.io/
+.. _SQR-049: https://sqr-049.lsst.io/
+.. _SQR-039: https://sqr-039.lsst.io/
+
+As much as possible, authentication for applications in the Rubin Science Platform will be handled at the ingress layer of Kubernetes, before the external request reaches the application.
+The underlying application can then be authentication-agnostic (if it doesn't need to make internal API calls on behalf of the user and doesn't need the user's identity), or can have a very simple authentication layer where it trusts user metadata and tokens that are passed to it in HTTP headers.
+
+`Gafaelfawr`_ implements this authentication layer and supports passing user metadata and delegated access tokens to applications via HTTP headers.
+
+Access control
+--------------
+
+Access control is done by scope.
+Each meaningful domain of access must be assigned a unique scope in the configuration of the authentication system.
```

Comment:
Not a real comment, but as an aside: allowed scoped may also very by time, for example if we are flooded with users and have limited resources we may say that "for the next 24 hours, only users of group X can have access to nublado". Just something to think about when we are thinking about (a) token renewal/expiration and (b) how we present authorsiation fail messages to the user

Reply:
Good point, I haven't been thinking about rate limiting that way but it's a good way to think about it. I don't think it fits into this document, but I'll keep it in mind.